src/modules/sequencer-crosshair/CrosshairsPlaceable.js

```diff
@@ -0,0 +1,319 @@
+import { get_mouse_position, get_object_position, } from "../../lib/canvas-lib.js";
+
+export default class CrosshairsPlaceable extends MeasuredTemplate {
+
+	#handlers = {
+		confirm: null,
+		cancel: null,
+		move: null,
+		wheel: null,
+	};
+
+	#promise = {
+		resolve: null,
+		reject: null,
+	};
+
+	#isDrag = false;
+
+	get crosshair() {
+		return this.document.crosshair;
+	}
+
+	get callbacks() {
+		return this.document.callbacks;
+	}
+
+	async draw() {
+		await super.draw();
+
+		this.controlIcon.renderable = this.crosshair.icon.display;
+		if (this.crosshair.icon.texture) {
+			this.controlIcon.icon.iconSrc = this.crosshair.icon.texture;
+			this.controlIcon.draw();
+		}
+
+		return this;
+	}
+
+	_refreshRulerText() {
+		this.ruler.renderable = this.crosshair.label.display;
+		if (!this.crosshair.label.display) return;
+		if (this.crosshair.label?.text) {
+			this.ruler.anchor.set(0.5, 2.25);
+			this.ruler.position.set(this.crosshair.label.dx ?? 0, this.crosshair.label.dy ?? 0)
+			this.ruler.text = this.crosshair.label?.text ?? "";
+		} else {
+			return super._refreshRulerText();
+		}
+	}
+
+	async show() {
+		await this.draw();
+		this.layer.addChild(this);
+		this.oldInteractiveChildren = this.layer.interactiveChildren;
+		this.layer.interactiveChildren = false;
+		this.updatePosition();
+		if (this.callbacks["show"]) {
+			this.callbacks["show"](this);
+		}
+		return this.activateShowListeners();
+	}
+
+	async activateShowListeners() {
+		return new Promise((resolve, reject) => {
+			this.#promise.resolve = resolve;
+			this.#promise.reject = reject;

```

Comment:
Not sure what level of compatability you're looking for but the pretty newly released `Promise.withResolvers()` makes this pattern easier imo!